src/cpp/src/continuous_batching_impl.cpp

```diff
@@ -284,6 +307,10 @@ bool ContinuousBatchingPipeline::ContinuousBatchingImpl::has_non_finished_reques
 }
 
 void ContinuousBatchingPipeline::ContinuousBatchingImpl::step() {
+    if (m_model_input_type == ModelInputType::EMBEDDINGS && m_scheduler->get_config().enable_prefix_caching) {
+        OPENVINO_THROW("Prefix caching is not supported for VLM models.");
+    }
```

Comment:
I would move such exception to a place where `restore_cached_blocks` is called:
https://github.com/openvinotoolkit/openvino.genai/blob/405d123dcd3a233b1a82c78a29df89b85bef573b/src/cpp/src/continuous_batching_impl.cpp#L257-L259

Reply:
Done.